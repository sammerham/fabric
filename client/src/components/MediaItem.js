import React from 'react'

import {
  StyledImage,
  StyledTitle,
  StyledYear,
  StyledType,
  StyledDivMedia
} from './StyledComponents';

const DEFAULT_POSTER = 'https://www.hopkinsmedicine.org/-/media/feature/noimageavailable.ashx';


/*
 * Media Item
 *
 * This is the page where you rednder all info about single item
 * State: None

 * props:
      *Item { "id": 1, "title": "The Matrix", "imdbid": "tt0133093", "media_type": "movie", "release_year": 1999}
  
 * App ---->> MediaList ----> MediaItem
 */

function MediaItem({ item }) {
  const { Title, Year, Poster, Type } = item;

  return (
    <StyledDivMedia>
      <StyledImage src={Poster !== "N/A" ? Poster:DEFAULT_POSTER} alt="media" />
      <StyledTitle bg={'#448565'}>{ Title}</StyledTitle>
      <StyledYear bg={'#58755b'}>Release Year: { Year}</ StyledYear>
      <StyledType bg={'#58755b'}>Media Type: { Type }</StyledType>
    </StyledDivMedia>
   
  )
}

export default MediaItem
