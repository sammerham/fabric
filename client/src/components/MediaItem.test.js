import React from 'react'
import { render, screen } from '@testing-library/react';
import MediaItem from './MediaItem';


let MOCK = {
  "Title": "The Matrix Reloaded test",
  "Year": "2003",
  "imdbID": "tt0234215",
  "Type": "movie",
  "Poster": "https://m.media-amazon.com/images/M/MV5BODE0MzZhZTgtYzkwYi00YmI5LThlZWYtOWRmNWE5ODk0NzMxXkEyXkFqcGdeQXVyNjU0OTQ0OTY@._V1_SX300.jpg"
}

// smoke test
it('renders without crashing', () => {
  render(<MediaItem item={MOCK}/>);
});



// snapShot
it("matches snapshot", function () {
  const { container } = render(<MediaItem item={MOCK}/>);
  expect(container).toMatchSnapshot();
});


test("renders the correct item", function () {
  render(<MediaItem item={MOCK} />);
  expect(screen.getByText("The Matrix Reloaded test")).toBeInTheDocument()
});