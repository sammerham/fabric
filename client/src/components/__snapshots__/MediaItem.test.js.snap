// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="sc-jRQBWg bdlPlA"
  >
    <img
      alt="media"
      class="sc-hKwDye iPhefi"
      src="https://m.media-amazon.com/images/M/MV5BODE0MzZhZTgtYzkwYi00YmI5LThlZWYtOWRmNWE5ODk0NzMxXkEyXkFqcGdeQXVyNjU0OTQ0OTY@._V1_SX300.jpg"
    />
    <h3
      class="sc-bdvvtL fWhxGB"
    >
      The Matrix Reloaded test
    </h3>
    <h6
      class="sc-gsDKAQ drtXCm"
    >
      Release Year: 
      2003
    </h6>
    <h6
      class="sc-dkPtRN hRrCKw"
    >
      Media Type: 
      movie
    </h6>
  </div>
</div>
`;
